### DISCUSSION TITLE
searx integration

### DISCUSSION THREAD

**Discussion by ghost** [searchmysite/searchmysite.net#60] [General]
Hi,

The code I've been using to get searchmysite.net results in searx is

```
  - name : searchmysite
    shortcut: sms 
    engine: xpath
    timeout : 3.0
    paging : True
    disabled : True
    search_url : https://searchmysite.net/search/?q={query}&p={pageno}
    url_xpath : //div[@class="search-result"]/a[@class="result-link"]
    title_xpath : //div[@class="search-result"]/h5/a[@class="result-title"]
    content_xpath : //div[@class="search-result"]
```

This pulls in the main result from each domain, but not the additional results from the same domain.

When the above was blocked earlier this month, pulling results using the API still worked, but as far as I can tell the API can only pull results for a single domain.  For example:
` search_url : https://searchmysite.net/api/v1/search/searchmysite.net?q={query}`
shows only results local to searchmysite.net, not general search results.  Is there a way to set the API to show results from all domains?

**Comment by m-i-l** [comment]
searx integration does sounds good, but AFAIK searx is more for the non-privacy respecting search engines, so it would be more of a convenience to have it in searx rather than improving your privacy or anything more fundamental like that (searchmysite.net has zero cookies on the main search pages, uses privacy aware analytics, has no incentive to collect user data for sale to advertisers, etc.)

On extending the API from single-site to all sites, a user did ask about that way back on 21 Oct 2020 (before it was on github.com), and I'll copy and paste my response because I think if anything it is even more relevant unfortunately: "I like the idea [of a site-wide API], and like microformats, structured data, nice APIs and so on. But I don't want to make it too easy to get all the data at the moment to be honest, because if it is too easy I have this fear someone will come along, rip it off, put it on some advert laden site, and use some blackhat SEO to get their site higher in Google."
